janis_bioinformatics/tools/gatk4/mutect2/base_4_1_cram.py

```diff
@@ -0,0 +1,27 @@
+from janis_bioinformatics.data_types import Bam, BamBai, Cram, CramCrai
+from janis_core import Array, String
+
+from .base_4_1 import Gatk4Mutect2Base_4_1
+
+
+class Gatk4Mutect2CramBase_4_1(Gatk4Mutect2Base_4_1):
```

Comment:
Cool!

Reply:
Hey @SebastianHollizeck, as I was runnings some tests, you don't respect whether an input was optional or not. You could get away with at the start, saying:

```python
for inp in ins:
    is_optional = inp.input_type.optional
    # checking code
    inp.input_type.optional = is_optional

return ins
```

This can be applied to all of your inputs.

You could probably standardise this by writing some function that does this transformation for you: `def transform_input_bams_to_crams(inputs): <implementation>`, then in each of your subclasses, just do something like:

```python
class MyCramSubclass(MyNormalTool)
    def inputs(self):
        return transform_input_bams_to_crams(super().inputs())
```